path_login.go

```diff
@@ -338,12 +343,12 @@ func (s *serviceAccount) namespace() string {
 }
 
 type projectedServiceToken struct {
-	Namespace      string                      `mapstructure:"namespace"`
-	Pod            *projectedServiceAccountPod `mapstructure:"pod"`
-	ServiceAccount *projectedServiceAccountPod `mapstructure:"serviceaccount"`
+	Namespace      string     `mapstructure:"namespace"`
+	Pod            *reference `mapstructure:"pod"`
+	ServiceAccount *reference `mapstructure:"serviceaccount"`
 }
 
-type projectedServiceAccountPod struct {
+type reference struct {
```

Comment:
nit: the name of this struct might be too generic,  I wonder if maybe `tokenMeta` could be any better?